src/storage/query/GetNeighborsProcessor.cpp

```diff
@@ -92,6 +93,19 @@ void GetNeighborsProcessor::runInSingleThread(const cpp2::GetNeighborsRequest& r
         onFinished();
         return;
       }
+      nebula::DataSet ds(*req.column_names_ref());
+      ds.rows.emplace_back(row);
+      size_t len = plan.getNodes().size();
+      for (size_t i =0; i < len; i ++)
+      {
+        RelNode<VertexID>* rn = plan.getNode(i);
+        std::string& na = const_cast<std::string&>(rn->name());
+        if(na == "FilterNode"){
+          FilterNode<VertexID>* der = dynamic_cast<FilterNode<VertexID>*>(rn);
+          der->reqDataSet = ds;
```

Comment:
We'd better set the the dataSet when we build the `FilterNode`, which is in function `buildPlan`

Reply:
The data in the filter is variable. Is it difficult to control the current buildPlan stage

Reply:
Just put the code into BuildPlan is enough?
```
      nebula::DataSet ds(*req.column_names_ref());
      ds.rows.emplace_back(row);
     
      ...
      make_unique<FilterNode> filter;
      filter->setReqDataSet(ds);
```